### PR TITLE
added filtering functionality using selectors

### PR DESCRIPTION
**Pull request by WGillian**

### CODE REVIEW

File: src/components/journeys/index.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ export default class Journeys extends React.Component {
           <h1>Your journeys.</h1>
           <Container>
             <Header>Future trips</Header>
+            <Input header="Filter trips" onChange={e => this.props.onFilterTrips(e.target.value)} value={this.props.filterText} placeholder="Search..." />
```
Comment (richardgillzdev) [inline src/components/journeys/index.js]: Propably `onUpdateFilterText`

File: src/containers/journeys/index.js [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
   render() {
     return (
       <Journeys
-        trips={this.props.trips}
+        trips={this.props.trips.filteredTrips}
+        filterText={this.props.trips.filterText}
         onAddTrip={trip => this.props.dispatch(addTrip(trip))}
         updateTrip={trip => this.props.dispatch(updateTrip(trip))}
         deleteTrip={tripId => this.props.dispatch(deleteTrip(tripId))}
+        onFilterTrips={filterText => this.props.dispatch(filterTrips(filterText))}
```
Comment (richardgillzdev) [inline src/containers/journeys/index.js]: `filterTrips` should probably be `updateFilterText` to make it more obvious what it does. Change this everywhere else as well.

File: src/redux/selectors/trips.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+export default createSelector(
+  'trips',
+  state => state.trips,
+  tripsState => {
+    const trips = tripsState.trips
+    const filter = tripsState.filterText
+    const filteredTrips = filterTrips(trips, filter)
+    return {
+      ...tripsState,
+      filteredTrips: filteredTrips,
```
Comment (richardgillzdev) [inline src/redux/selectors/trips.js]: Just call this property `trips`

File: src/redux/selectors/trips.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  }
+  return _.filter(trips, trip => trip.destination.includes(filterSearch))
+}
+
+export default createSelector(
+  'trips',
+  state => state.trips,
+  tripsState => {
+    const trips = tripsState.trips
+    const filter = tripsState.filterText
```
Comment (richardgillzdev) [inline src/redux/selectors/trips.js]: `trips` and `filter` proably don't need to be assigned to const's

File: src/containers/journeys/index.js (hunk 2 above)
Comment (richardgillzdev) [reply src/containers/journeys/index.js]: and 'onFilterTrips' should be `onUpdateFilterText`

File: src/components/journeys/index.js (hunk 1 above)
Comment (WGillian) [reply src/components/journeys/index.js]: My `<Input />` component in hooked up to watch for `this.props.onChange` and is being used elsewhere. Should I create a new `<FilterInput />` component?

Comment (richardgillzdev) [reply src/components/journeys/index.js]: Sorry, what I meant was `onChange={e => this.props.onUpdateFilterText(e.target.value)}`

I don't think `onFilterTrips` is a good description of what it is doing, this should be changed in all the code

File: src/containers/journeys/index.js [hunk 5]
```diff
@@ -1,18 +1,22 @@
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
 import Journeys from 'components/journeys'
-import { addTrip, updateTrip, deleteTrip } from 'redux/actions/trips'
+import { addTrip, updateTrip, deleteTrip, updateFilterText } from 'redux/actions/trips'
+import tripsSelector from '../../redux/selectors/trips'
 
 class JourneysContainer extends Component {
   render() {
+    console.log(this.props)
```
Comment (richardgillzdev) [inline src/containers/journeys/index.js]: Remove

File: src/redux/selectors/trips.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    return trips
+  }
+  return _.filter(trips, trip => trip.destination.includes(filterSearch))
+}
+
+export default createSelector(
+  'trips',
+  state => state.trips,
+  tripsState => {
+    const filteredTrips = filterTrips(tripsState.trips, tripsState.filterText)
```
Comment (richardgillzdev) [inline src/redux/selectors/trips.js]: You probably don't even need to assign this to a value now either

**Pull request merged**